projects/mercury/src/llm/Conversation.js

```diff
@@ -0,0 +1,415 @@
+        if (data && data.length > 0) {
+            data.sort((a, b) => new Date(b.start) - new Date(a.start));
+
+            if (JSON.stringify(data) !== JSON.stringify(conversationHistory)) {
+                setConversationHistory(data);
+            }
+        }
+    };
+
+    const getAllConversationHistory = () => {
+        new FulltextAPI()
+            .getAllConversations()
+            .then(processConversationHistory)
+            .catch(() => {
+                handleHttpError('Error retrieving chat history.');
+                setLoading(false);
+            });
+    };
+
+    const prepareRestoreChat = id => {
+        setLoading(true);
+        setQuery('');
+        setResponseArticles([]);
+        setMessages([]);
+        setConversationId(id);
+        setRestoreChatStatus(true);
+        getAllConversationHistory();
+    };
+
+    const startNewConversation = () => {
+        setQuery('');
+        setResponseMessage('');
+        setMessages([]);
+        setResponseArticles([]);
+        setConversationId('');
+        getAllConversationHistory();
+    };
+
+    const deleteChat = id => {
+        new FulltextAPI()
+            .deleteChat(id)
+            .then(getAllConversationHistory)
+            .then(startNewConversation)
+            .then(() => setLoading(false))
+            .catch(() => handleHttpError('Error deleting chat history with id ' + id));
+    };
+
+    const processSearchQueryChange = newQuery => {
+        if (newQuery !== query) {
+            if (responseMessage !== '') {
+                setResponseMessage('');
+                setResponseArticles([]);
+            }
+
+            setQuery(newQuery);
+        }
+    };
+
+    const handleSearchError = error => {
+        setResponseMessage(error);
+        setLoading(false);
+    };
+
+    const performSearch = () => {
+        if (conversationId === '') {
+            const searchResult = new FulltextAPI().search(query);
+            return searchResult;
+        }
+
+        return new FulltextAPI().chat(query, conversationId);
+    };
+
+    const prepareFetchSearch = () => {
+        setLoading(true);
+        setResponseMessage('');
+
+        if (query === '') {
+            return;
+        }
+
+        performSearch()
+            .then(processResponseMessages)
+            .then(() => setLoading(false))
+            .catch(() => {
+                handleHttpError('Connection error.');
+                handleSearchError('Fairspace article search is not available at the moment');
+            });
+    };
+
+    const extractArticleIri = webResult => {
+        if (webResult && webResult.length > 0) {
+            setArticleIri(webResult[0].id);
+        } else {
+            setArticleIri('');
+        }
+    };
+
+    const showArticle = articleId => {
+        if (articleId !== null && articleId.length > 0) {
+            LocalSearchAPI.lookupSearch(articleId, 'https://www.fns-cloud.eu/study').then(extractArticleIri);
```

Comment:
I don't like hardcoded prefixes in the generic UI, this is a component that works in a really specific setup...